src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -34,6 +34,10 @@
 //      assert false;
 //  }
 
+    private void failAssertionErrorExpected() {
+        throw new AssertionError("AssertionError expected");
```

Comment:
I think I would prefer to extract a static constant for "AssertionError expected" and inline this method. That's because it's 100% clear that this code is wrong:

    try {
      fail("woops');
      throw new AssertionError(ASSERTION_ERROR_EXPECTED);
    } catch (AssertionError expected) {
      ...